tests/jballerina-unit-test/src/test/resources/test-src/types/table/table_value_assign_test.bal

```diff
@@ -144,12 +149,38 @@ function testAssignKeylessTableValueToTableType() {
     ];
     var a2 = table key() from var {id} in tbl1 select {id};
     table<record {| int id; |}> a3 = a2;
-    any a4 = table [{"id": 1234}, {"id": 4567}];
-
-    assertEqual(table [{"id": 1234}, {"id": 4567}], a3);
+    var a4 = table [{"id": 1234}, {"id": 4567}];
     assertEqual(a3, a4);
 }
 
+function testTableVlueAssignmentToAny() {
+    any tbl = table[];
+    table<map<any|error>> a1 = <table<map<any|error>>> tbl;
+    a1.add({"any": "any"});
+    a1.add({"error": error("")});
+
+    any t2 = table [{a: 1}];
+    table<map<any|error>> t3 = <table<map<any|error>>> t2;
+    map<any|error> m = {a: error("")};
+    t3.add(m);
+
```

Comment:
Shall we add a few tests with `any` with table-constructor-expr having a key-specifier too?

Reply:
added tests, according to https://github.com/ballerina-platform/ballerina-spec/issues/1092 it has to be a compile error. So added negative tests